src/arrays/arraymath.jl

```diff
@@ -34,29 +35,57 @@ function *{B<:OrthonormalBasis}(op::DualMatrix{B}, ket::QuVector{B})
 end
 
 # ket * bra -> operator
-function *{B<:OrthonormalBasis}(ket::QuVector{B}, bra::DualVector{B}) 
-    return QuArray(A_mul_Bc(rawcoeffs(ket), rawcoeffs(bra)), 
-                   bases(ket,1), 
+function *{B<:OrthonormalBasis}(ket::QuVector{B}, bra::DualVector{B})
+    return QuArray(A_mul_Bc(rawcoeffs(ket), rawcoeffs(bra)),
+                   bases(ket,1),
                    bases(bra,1))
 end
 
 # operator * operator -> operator
 function *{B<:OrthonormalBasis}(qm::QuMatrix{B}, dm::DualMatrix{B})
-    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)), 
-                   bases(qm,1), 
+    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)),
+                   bases(qm,1),
                    bases(dm,2))
 end
 
 function *{B<:OrthonormalBasis}(dm::DualMatrix{B}, qm::QuMatrix{B})
-    return QuArray(Ac_mul_B(rawcoeffs(dm), rawcoeffs(qm)), 
-                   bases(dm,1), 
+    return QuArray(Ac_mul_B(rawcoeffs(dm), rawcoeffs(qm)),
+                   bases(dm,1),
                    bases(qm,2))
 end
 
-function *{B<:OrthonormalBasis}(qm1::QuMatrix{B}, qm2::QuMatrix{B}) 
-    return QuArray(rawcoeffs(qm1)*rawcoeffs(qm2), 
-                   bases(qm1,1), 
+function *{B<:OrthonormalBasis}(qm1::QuMatrix{B}, qm2::QuMatrix{B})
+    return QuArray(rawcoeffs(qm1)*rawcoeffs(qm2),
+                   bases(qm1,1),
                    bases(qm2,2))
 end
 
 *(dm1::DualMatrix, dm2::DualMatrix) = (dm1.qarr*dm2.qarr)'
+*(num::Number, qarr::QuArray) = scale(num, qarr)
+/(qarr::QuArray, num::Number) = scale(1/num, qarr)
+
+function scale(num::Number, qarr::QuArray)
+    return scale!(num, copy(qarr))
+end
+
+function scale!(num::Number, qarr::QuArray)
+    scale!(num, qarr.coeffs)
```

Comment:
Why not `rawcoeffs(qarr)` instead of `qarr.coeffs`? As I said earlier, if you use `rawcoeffs` you can change `qarr::QuArray` to `qarr::Union(QuArray,CTranspose)` and your functions will work for both types. Does that not work?